### PR TITLE
fix/ KuCoinAPIOrderBookDataSource subscribing to all active markets

### LINKED ISSUE #3223: Kucoin - Memory leak when running on development
**Describe the bug**
When running the Kucoin connector on the latest development, there's a significant increase in memory usage in a short period of time. So far I was able to run a pure_mm and avellaneda_mm but got the same result with memory usage. I'm also seeing these errors after stopping the strategy, similar to this [bug](https://github.com/CoinAlpha/hummingbot/issues/2627)

Screen Shot 2021-04-18 at 1 35 00 PM

Full traceback error can be seen on the attached log file below.

**Steps To Reproduce**

1. Run a pure_mm or avellaneda_mm using Kucoin as the connector.
2. After running, check for the memory usage using `htop` on your terminal.
3. Notice the increase in memory usage in a short period of time (CPU%, Mem and RES).
4. Cross checking with other connectors, I was only able to observe this behavior on Kucoin.

**Screenshots**
Attached screenshot and clips are shortly after running the strategy (pure_mm), for the samples on the master branch, I only attached an image since memory was stable upon running for quite some time.

Installation: Docker
Branch: Master
kucoin_docker_master_pure_mm

Installation: Docker
Branch: Development

https://user-images.githubusercontent.com/64377055/115136001-5bec3780-a04f-11eb-8dd6-5d2fe83a41e6.mp4

Installation: Source
Branch: Master
kucoin_pure_mm_master_source

Installation: Source
Branch: Development

https://user-images.githubusercontent.com/64377055/115136032-92c24d80-a04f-11eb-925d-40bd3ad4090c.mp4

**Release version**
```
branch: development
commit: 9e87685d54e344d4ee372db5e0baefa1644139f9
date: Fri Apr 16 16:43:59 2021 +0800
```

**Attachments**
Logs: [logs_kucoin_eth-usdt.log](https://github.com/CoinAlpha/hummingbot/files/6330576/logs_kucoin_eth-usdt.log)
Config: [kucoin_eth-usdt.txt](https://github.com/CoinAlpha/hummingbot/files/6330577/kucoin_eth-usdt.txt)

### PR DESCRIPTION
**Pull request by CrimsonJacket**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Possibly fixes #3223 

- ~~Initial investigation using `objgraph` seems to show that there might be a "rogue" function creating an excessive no. of `Task`~~ It appears that these `Task` are the ping tasks that is created to maintain the websocket connection.
image

- Further examining the code, it appears that the connector subscribes to all the active markets available. This seems to be an entirely separate issue, but nonetheless fixing this has shown a considerable decrease in CPU and RAM usage.
image

- This fix does not necessarily resolve the memory leak issue, but this addresses it temporarily.

### CODE REVIEW

**Comment by R-K-H** [comment]
Instead of all your conditional checks, it might be best to instead `await get_trading_pairs` as it essentially does the same thing?

File: hummingbot/connector/exchange/kucoin/kucoin_api_order_book_data_source.py [hunk 1]
```diff
@@ -345,7 +347,7 @@ async def get_new_order_book(self, trading_pair: str) -> OrderBook:
 
     async def get_markets_per_ws_connection(self) -> List[str]:
         # Fetch the  markets and split per connection
-        all_symbols: List[str] = await self.fetch_trading_pairs()
+        all_symbols: List[str] = self._trading_pairs if self._trading_pairs else await self.fetch_trading_pairs()
```
Comment (R-K-H) [inline hummingbot/connector/exchange/kucoin/kucoin_api_order_book_data_source.py:350]: In place of this check, replace `self.fetch_trading_pairs()` with `self.get_trading_pairs()` which does the same check as this if and keeps the checks in the correct place?

Comment (R-K-H) [reply hummingbot/connector/exchange/kucoin/kucoin_api_order_book_data_source.py:350]: So as you don't need the `self._trading_pairs if self._trading_pairs else`

**Comment by RC-13** [comment]
Run a test bot for more than 24 hours now and memory usage stayed at 400 MB+ with multiple instances running. LGTM

**Review by Nullably — APPROVED** [review]
LGTM

**Pull request merged**